### DISCUSSION TITLE
Division by zero error for node classification

### DISCUSSION THREAD

**Discussion by AdarshMJ** [pyg-team/pytorch_geometric#5252] [Q&A] [answered]
Hi @rusty1s 

I am trying to run a simple GCN on Cornell dataset(WebKB). My model looks like this -
```
GCN(
  (conv1): GCNConv(1703, 32)
  (conv2): GCNConv(32, 5)
)
```
The dataset looks like this originally -

`Data(x=[183, 1703], edge_index=[2, 298], y=[183], train_mask=[183, 10], val_mask=[183, 10], test_mask=[183, 10])`

I do some modifications by converting it to NetworkX graph and then convert it back to PyG graph, to ensure I get the masks back I apply the random node split transform -

```
Newdatapyg = from_networkx(gcornell)
from torch_geometric.transforms import RandomNodeSplit
transform = RandomNodeSplit(split='random')
Newdatapyg  = transform(Newdatapyg)
print(Newdatapyg)
```

I get -

`Data(x=[183, 1703], edge_index=[2, 561], y=[183], train_mask=[183], val_mask=[183], test_mask=[183])
`
When I train this, my accuracy looks like this -

![Screenshot 2022-08-20 at 12 04 15](https://user-images.githubusercontent.com/24205200/185740692-6a8a459a-74d6-4795-96b2-d4e775ad0b05.png)

But when I try to test it I get the following error -

![Screenshot 2022-08-20 at 12 04 55](https://user-images.githubusercontent.com/24205200/185740738-3cf3e442-5468-4a77-933d-a74197a1e104.png)

Turns out in this line - `test_acc = int(test_correct.sum()) / int(Newdatapyg.test_mask.sum())` the denominator is 0. Is it the problem of the model? Is it bad that its not able to work on test set? Or am I doing something wrong? Thanks for your time :)

**Comment by Padarn** [comment, marked as answer]
Are you sure that there are any nodes in the test set? The `test_mask` could be full of zeros, making the denominator zero.

**Reply by AdarshMJ** [reply]
Is it possible because of the way the `test_mask` looks? If i print -

`Data(x=[183, 1703], edge_index=[2, 298], y=[183], train_mask=[183], val_mask=[183], test_mask=[183])`

`print(Newdatapyg.test_mask)`

```
I get -
tensor([False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False])
```

`print(Newdatapyg.test_mask.sum())`
I get -` tensor(0)`

How to train a model if all the test_mask consists of all False?

**Reply by Padarn** [reply]
Yes then this is the cause.

You can still train the model you'll just need to remove the `test_acc` (which is meaningless as you have no test cases). It looks like this is from code you wrote  so it should be simple to remove?

**Reply by AdarshMJ** [reply]
Yes. Thank you :) I do have val_mask which are not all False. So I can use that instead of test_mask.
